src/pages/transactions/addTransaction.js

```diff
@@ -0,0 +1,72 @@
+import { useRouter } from "next/router";
+
+export default function TransactionForm({ onAddTransaction, onCancel }) {
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    onAddTransaction(data);
+    router.push("/");
+  }
+
+  return (
+    <div>
+      <h1>Expense</h1>
+      <form onSubmit={handleSubmit}>
+        <button type="button" onClick={onCancel}>
+          Cancel
+        </button>
+        <div>
+          <label htmlFor={"type"}>Transaction Type:</label>
```

Comment:
htmlFor={"type"} not need as it is a string `htmlFor="type"` should do the job.
consider the change overall